### ISSUE TITLE
Add rule to report relative parent paths

### ISSUE THREAD

**Issue by FloEdelmann** [import-js/eslint-plugin-import#2616]
There is the `import/no-relative-parent-imports` rule which reports all imports from parent folders, even if you use an absolute path (e.g. with [webpack aliases](https://webpack.js.org/configuration/resolve/#resolvealias) or [TypeScript path mapping](https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping)).

```
root
├─ foo
│  └─ bar
│     └─ baz
│        └─ child-module.ts
└─ parent-module.ts
```

```ts
// foo/bar/baz/child-module.ts

import '../../../parent-module'; // reported
import '@/parent-module'; // also reported
```

I want to only disallow unreadable parent imports, i.e. those with more than two `../` in their path, but still allow importing the same module with an absolute path:

```ts
// foo/bar/baz/child-module.ts

import '../../../parent-module'; // should be reported (more than two `../`)
import '@/parent-module'; // should NOT be reported
```

I'm not sure if we can achieve this via options in the `import/no-relative-parent-imports` or if this should be a new separate rule.

**Comment by ljharb** [comment]
Why is "more than two `../`" notable here?

**Comment by FloEdelmann** [comment]
Because I'd like to have that configurable. One or two `../` would be fine for me, I'd only like to forbid more than two `../` in a path.

**Comment by ljharb** [comment]
Why?

**Comment by FloEdelmann** [comment]
* `../foo` is readable
* `../../foo` is still relatively readable
* `../../../foo` already requires too much thinking about in which folder to search for the file, so it should better be rewritten to an absolute path

The threshold might be different for other people/projects, so I'd make that configurable in the rule (either the existing `import/no-relative-parent-imports` rule or a new rule).
